dbt_copilot_helper/utils/validation.py

```diff
@@ -142,3 +199,197 @@ def validator(string):
         ],
     },
 )
+
+NUMBER = Or(int, float)
+DELETION_POLICY = Or("Delete", "Retain", "Snapshot")
```

Comment:
The "Snapshot" deletion policy is only applicable to redis and postgres addons. It's still valid Cloudformation but defaults to delete if the resource doesn't support it. Might be worth protecting against any assumptions that snapshot would work for s3/opensearch?

```suggestion
DELETION_POLICY = Or("Delete", "Retain")
DELETION_POLICY_DATABASE = Or("Delete", "Retain", "Snapshot")
```